src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/EmbeddingProviderFactory.java

```diff
@@ -96,33 +95,29 @@ private synchronized EmbeddingProvider addService(
     }
 
     if (configuration.serviceProvider().equals(ProviderConstants.CUSTOM)) {
+      Optional<Class<?>> clazz = configuration.implementationClass();
+      if (!clazz.isPresent()) {
+        throw ErrorCode.VECTORIZE_SERVICE_TYPE_UNAVAILABLE.toApiException("custom class undefined");
+      }
       try {
-        Optional<Class<?>> clazz = configuration.implementationClass();
-        if (clazz.isPresent()) {
-          final EmbeddingProvider customEmbeddingProvider =
-              (EmbeddingProvider) clazz.get().getConstructor().newInstance();
-          return customEmbeddingProvider;
-        } else {
-          throw new JsonApiException(
-              ErrorCode.VECTORIZE_SERVICE_TYPE_UNAVAILABLE,
-              ErrorCode.VECTORIZE_SERVICE_TYPE_UNAVAILABLE.getMessage() + "custom class undefined");
-        }
+        return (EmbeddingProvider) clazz.get().getConstructor().newInstance();
       } catch (Exception e) {
-        throw new JsonApiException(
```

Comment:
Refactored so that "JsonApiException" just thrown is not caught, wrapped, re-thrown.